### PR TITLE
Documentation/design: add prepare design

### PR DESCRIPTION
**Pull request by abhinavdahiya**
Add design doc for prepare step.

### CODE REVIEW

File: Documentation/design/preparephase.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+2. Kubernetes objects to be installed in the cluster.
+
+The type of additions not supported are:
+
+1. Assets like systemd-service files for master and worker nodes.
+
+### Output
+
+The Prepare step creates ignition file for bootstrap, master and worker node namely, `bootstrap.ign`, `master.ign` and `worker.ign` in the current working directory or the directory specified by `--assets-dir` flag.
+
```
Comment (yifan-gu) [inline Documentation/design/preparephase.md]: maybe also add `Example prepare output`? Also indicate where users can put their customized files?

Comment (abhinavdahiya) [reply Documentation/design/preparephase.md]: done.

**Pull request merged**